src/buza/managers.py

```diff
@@ -0,0 +1,59 @@
+import re
+
+from django.contrib.auth.base_user import BaseUserManager
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+
+
+def validate_phone(phone):
+    if phone.startswith("+27"):
+        phone_pattern = re.compile(r'[+]27(\d{9})$')
+        return phone_pattern.match(phone)
+    else:
+        phone_pattern = re.compile(r'0(\d{9})$')
+        return phone_pattern.match(phone)
+
+
+class UserManager(BaseUserManager):
```

Comment:
This should probably be in `models.py`, rather than in a separate module. (That tends to create import loops, for code this closely related.)

We should also see if we can get away without adding a custom manager, at first: we don't necessarily need it, and that can simplifies things.